### DISCUSSION TITLE
judge_server and judger is empty

### DISCUSSION THREAD

**Discussion by hamedmax73** [ZsgsDesign/NOJ#535] [Q&A] [answered]
hi
I install NOJ and it's very good. after that, I installed bable extensions NOJ, ZOJ,...
but none of them is working
I checked the database and judge_server and judger is empty is empty
at submitter.php and line 29 
 $this->oid=OJModel::oid('noj');
 return always null 
 
 please help me what is rung
I check all documentation and search the entire internet but I didn't found any anwer

**Comment by ZsgsDesign** [comment, marked as answer]
Did you run `php artisan babel:install noj` and others? If so `$this->oid=OJModel::oid('noj');` would not return `null`, it will return an integer indicate the oid of that online judge.

You can check the database table `OJ` for that.

After installed certain OJ, you will need to configure the `judger` and `judge_server` table. Judge Server table requires you input NOJ_JudgeServer hosts and ports, token. you will also need to name that server though.

For virtual judge, you will need to insert accounts into `judger` table manually, it will require you input password and login user into it.

You can post issues and we will reply to you much quicker, we didn't quite understand this new discussion feature for now.
